dependencies.go

```diff
@@ -0,0 +1,47 @@
+package wal
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+)
+
+// These interfaces define the wal's dependencies. Using the smallest
+// interface possible makes it easier to mock these dependencies in testing.
+type (
+	dependencies interface {
+		disrupt(string) bool
+		readFile(string) ([]byte, error)
+		openFile(string, int, os.FileMode) (file, error)
+		create(string) (file, error)
+	}
+
+	// file implements all of the methods that can be called on an os.File.
+	file interface {
+		io.ReadWriteCloser
+		Name() string
+		ReadAt([]byte, int64) (int, error)
+		Sync() error
+		WriteAt([]byte, int64) (int, error)
+	}
+)
+
+type prodDependencies struct{}
```

Comment:
Should have a brief comment saying that prodDependencies is a passthorugh to the standard library